app/validators/value_source_validator.py

```diff
@@ -7,16 +9,43 @@ class ValueSourceValidator(Validator):
     SOURCE_REFERENCE_INVALID = "Invalid {} source reference"
     ANSWER_SOURCE_REFERENCE_INVALID = SOURCE_REFERENCE_INVALID.format("answers")
 
+    # Progress source reference errors
+    SOURCE_REFERENCE_CURENT_BLOCK = (
+        "Invalid {} source reference: cannot be current block"
+    )
+    SOURCE_REFERENCE_CURENT_SECTION = (
+        "Invalid {} source reference: cannot be current section"
+    )
+    SOURCE_REFERENCE_FUTURE_BLOCK = (
+        "Invalid {} source reference: cannot be a later block in the questionnaire"
```

Comment:
Minor - I would consider making these error messages more explicit, something like `The identifier being referenced in the progress source must come before the current block`?

Reply:
Changed all the messages according to suggestion